lib/Mojo/File.pm

```diff
@@ -0,0 +1,377 @@
+
+=head1 DESCRIPTION
+
+L<Mojo::File> is a scalar-based container for file system paths that provides a
+friendly API for dealing with different operating systems.
+
+  # Access scalar directly to manipulate path
+  my $path = Mojo::File->new('/home/sri/test');
+  $$path .= '.txt';
+
+=head1 FUNCTIONS
+
+L<Mojo::File> implements the following functions, which can be imported
+individually.
+
+=head2 path
+
+  my $path = path;
+  my $path = path('/home/sri/.vimrc');
+  my $path = path('/home', 'sri', '.vimrc');
+
+Construct a new scalar-based L<Mojo::File> object, defaults to using the current
+working directory.
+
+=head2 tempdir
+
+  my $path = tempdir;
+  my $path = tempdir('tempXXXXX');
+
+Construct a new scalar-based L<Mojo::File> object for a temporary directory with
+L<File::Temp>.
+
+  # Longer version
+  my $path = Mojo::File->new(File::Temp->newdir('tempXXXXX'));
+
+=head1 METHODS
+
+L<Mojo::File> implements the following methods.
+
+=head2 basename
+
+  my $name = $path->basename;
+  my $name = $path->basename('.txt');
+
+Return the last level of the path with L<File::Basename>.
+
+  # ".vimrc" (on UNIX)
+  Mojo::File->new('/home/sri/.vimrc')->basename;
+
+  # "test" (on UNIX)
+  Mojo::File->new('/home/sri/test.txt')->basename('.txt');
+
+=head2 child
+
+  my $child = $path->child('.vimrc');
+
+Return a new L<Mojo::File> object relative to the path.
+
+  # "/home/sri/.vimrc" (on UNIX)
+  Mojo::File->new('/home')->child('sri', '.vimrc');
+
+=head2 dirname
+
+  my $name = $path->dirname;
+
+Return all but the last level of the path with L<File::Basename> as a
+L<Mojo::File> object.
+
+  # "/home/sri" (on UNIX)
+  Mojo::File->new('/home/sri/.vimrc')->dirname;
+
+=head2 is_abs
+
+  my $bool = $path->is_abs;
+
+Check if the path is absolute.
+
+  # True (on UNIX)
+  Mojo::File->new('/home/sri/.vimrc')->is_abs;
+
+  # False (on UNIX)
+  Mojo::File->new('.vimrc')->is_abs;
+
+=head2 list
+
+  my $collection = $path->list;
+  my $collection = $path->list({hidden => 1});
+
+List all files in the directory and return a L<Mojo::Collection> object
+containing the results as L<Mojo::File> objects.
+
+  # List files
+  say for Mojo::File->new('/home/sri/myapp')->list->each;
+
+These options are currently available:
+
+=over 2
+
+=item dir
+
+  dir => 1
+
+Include directories.
+
+=item hidden
+
+  hidden => 1
+
+Include hidden files.
+
+=back
+
+=head2 list_tree
+
+  my $collection = $path->list_tree;
+  my $collection = $path->list_tree({hidden => 1});
+
+List all files recursively in the directory and return a L<Mojo::Collection>
+object containing the results as L<Mojo::File> objects.
+
+  # List all templates
+  say for Mojo::File->new('/home/sri/myapp/templates')->list_tree->each;
+
+These options are currently available:
+
+=over 2
+
+=item hidden
+
+  hidden => 1
+
+Include hidden files and directories.
+
+=back
+
+=head2 make_path
+
+  $path = $path->make_path;
+
+Create the directories if they don't already exist with L<File::Path>.
+
```

Comment:
A clean proposal with documentation and tests could convince me to support it though.